### PR TITLE
KAA-1133: Your first Kaa application - reviewed.

### PR DESCRIPTION
**Pull request by hunch7**

### CODE REVIEW

File: doc/Programming-guide/Your-first-Kaa-application/index.md [hunk 1]
```diff
@@ -10,149 +10,159 @@ sort_idx: 10
 * TOC
 {:toc}
 
-By walking through this guide, you will learn essential skills of creating Kaa-based applications, as well as the understanding of the Kaa approach in general.
-We'll cover here basic yet realistic example of collecting data from the sensor devices.
-For more advanced Kaa features, refer to the [next steps section](#next-steps).
+This section discusses the basics of creating Kaa-based applications using the [Kaa Sandbox]({{root_url}}Glossary/kaa-sandbox).
```
Comment (rasendubi) [inline doc/Programming-guide/Your-first-Kaa-application/index.md]: Link is broken

Comment (hunch7) [reply doc/Programming-guide/Your-first-Kaa-application/index.md]: Fixed.

**Comment by vadimol** [comment]
@kaa-leeroy re-test this please.

File: doc/Programming-guide/Your-first-Kaa-application/index.md [hunk 2]
```diff
@@ -165,39 +175,49 @@ The following code block illustrates a simple desktop application that sends vir
...
 
-        sudo apt-get install cmake
-1. Create `kaa` directory and unpack C SDK
-1. Create `CMakeLists.txt` in the application directory with following contents.
+		sudo apt-get install cmake
+		sudo apt-get install libssl-dev
 
-        cmake_minimum_required(VERSION 2.8.8)
-        project(kaa-application C)
+2. Create a direcory named **kaa** and unpack the C SDK archive it it.
```
Comment (vadimol) [inline doc/Programming-guide/Your-first-Kaa-application/index.md]: mistype: it it

File: doc/Programming-guide/Your-first-Kaa-application/index.md [hunk 3]
```diff
@@ -165,39 +175,49 @@ The following code block illustrates a simple desktop application that sends vir
...
 
-        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -g -Wall -Wextra")
+3. In the application directory, create a CMakeLists.txt file with the following contents.
 
-        add_subdirectory(kaa)
+		cmake_minimum_required(VERSION 2.8.8)
+		project(kaa-application C)
+		
+		find_package(OpenSSL REQUIRED)
```
Comment (vadimol) [inline doc/Programming-guide/Your-first-Kaa-application/index.md]: remove this line please, we do not need OpenSSL any more

File: doc/Programming-guide/Your-first-Kaa-application/index.md [hunk 4]
```diff
@@ -165,39 +175,49 @@ The following code block illustrates a simple desktop application that sends vir
...
+		
+		set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -g -Wall -Wextra")
+		
+		add_subdirectory(kaa)
+		
+		add_executable(kaa-app main.c)
+		
+		target_link_libraries(kaa-app kaac crypto)
```
Comment (vadimol) [inline doc/Programming-guide/Your-first-Kaa-application/index.md]: please, change to
target_link_libraries(kaa-app kaac)

**Review by vadimol — CHANGES_REQUESTED** [review]
Few minor comment were added. Please, take a look. 
Overall, looks good to me.

Regarding source code for C/C++, we will retest and update this guide anyway, so do not pay too much attention.

File: doc/Programming-guide/Your-first-Kaa-application/index.md (hunk 2 above)
Comment (hunch7) [reply doc/Programming-guide/Your-first-Kaa-application/index.md]: Fixed.

File: doc/Programming-guide/Your-first-Kaa-application/index.md (hunk 3 above)
Comment (hunch7) [reply doc/Programming-guide/Your-first-Kaa-application/index.md]: Done.

File: doc/Programming-guide/Your-first-Kaa-application/index.md (hunk 4 above)
Comment (hunch7) [reply doc/Programming-guide/Your-first-Kaa-application/index.md]: Done.

File: doc/Programming-guide/Your-first-Kaa-application/index.md [hunk 5]
```diff
@@ -165,39 +175,47 @@ The following code block illustrates a simple desktop application that sends vir
...
 
-1. Install dependencies: CMake. On Ubuntu, you can install it using following commands.
+1. Install CMake and OpenSSL.
+To do this, run the following commands in the terminal.
 
-        sudo apt-get install cmake
-1. Create `kaa` directory and unpack C SDK
-1. Create `CMakeLists.txt` in the application directory with following contents.
+		sudo apt-get install cmake
+		sudo apt-get install libssl-dev
```
Comment (rasendubi) [inline doc/Programming-guide/Your-first-Kaa-application/index.md]: We also don't need this line.

File: doc/Programming-guide/Your-first-Kaa-application/index.md [hunk 6]
```diff
@@ -165,39 +175,47 @@ The following code block illustrates a simple desktop application that sends vir
 
 <div id="prep-c" class="tab-pane fade in active" markdown="1" >
 
-Before you start with C application code, some preparation is required.
+Before using the C application code, be sure to complete the following steps:
 
-1. Install dependencies: CMake. On Ubuntu, you can install it using following commands.
+1. Install CMake and OpenSSL.
```
Comment (rasendubi) [inline doc/Programming-guide/Your-first-Kaa-application/index.md]: OpenSSL is not needed.

Comment (hunch7) [reply doc/Programming-guide/Your-first-Kaa-application/index.md]: Fixed.

File: doc/Programming-guide/Your-first-Kaa-application/index.md (hunk 5 above)
Comment (hunch7) [reply doc/Programming-guide/Your-first-Kaa-application/index.md]: Fixed.

**Review by vadimol — APPROVED** [review]

**Comment by vadimol** [comment]
@hunch7 , please resolve merge conflicts and assign the Jira issue to me.

**Comment by hunch7** [comment]
New input data added by developer, need to review again. Will make a new branch and a new PR. Closing this PR.

**Comment by hunch7** [comment]
New PR: https://github.com/kaaproject/kaa/pull/1093

**Pull request closed without merging**